sunpy/coordinates/frames.py

```diff
@@ -579,15 +579,27 @@ def make_3d(self):
         with np.errstate(invalid='ignore'):
             d = ((-1*b) - np.sqrt(b**2 - 4*c)) / 2  # use the "near" solution
 
-        if self._spherical_screen:
-            sphere_center = self._spherical_screen['center'].transform_to(self).cartesian
-            c = sphere_center.norm()**2 - self._spherical_screen['radius']**2
-            b = -2 * sphere_center.dot(rep)
-            # Ignore sqrt of NaNs
-            with np.errstate(invalid='ignore'):
-                dd = ((-1*b) + np.sqrt(b**2 - 4*c)) / 2  # use the "far" solution
+        if self._assumed_screen:
+            if self._assumed_screen['type'] == 'spherical':
```

Comment:
Instead of storing the name/ID of the screen and logic for doing the transform in separate places (the name in `self._assumed_screen` and the logic here), I think it would be cleaner to create a new `Screen` class that combines the two. Then `self._assumed_screen: Screen`, and `Screen` would have a `calculate_distance(self, ...)` method that would be called here. That would also make it neater to add other screens in the future if needed.